internal/pkg/service/buffer/sink/tablesink/storage/level/local/slice.go

```diff
@@ -35,9 +35,9 @@ func (f File) NewSlice(sliceDir string, previousSliceSize datasize.ByteSize) (Sl
 	var allocatedDiskSpace datasize.ByteSize
 	if f.DiskAllocation.Enabled {
 		if f.DiskAllocation.SizePercent > 0 && previousSliceSize > 0 {
-			allocatedDiskSpace = previousSliceSize
+			allocatedDiskSpace = (previousSliceSize * datasize.ByteSize(f.DiskAllocation.SizePercent)) / 100
 		} else {
-			allocatedDiskSpace = (f.DiskAllocation.Size * datasize.ByteSize(f.DiskAllocation.SizePercent)) / 100
+			allocatedDiskSpace = f.DiskAllocation.Size
 		}
```

Comment:
Ths was incorrect.

If we do not know the previous size (e.g. it is the first slice), then we use the static value from the configuration.
It is covered by the repository test, see previous commit.

Reply:
:)